joy/joy_node.cpp

```diff
@@ -150,10 +150,20 @@ int main(int argc, char * argv[])
       else
         break; // if we timed out, let ROS spin to do its stuff
     }
-    bool deadman = msg->buttons[0] != 0;
-    cmd_vel_msg->linear.x = deadman ? -msg->axes[1] * g_scale_linear : 0;
-    cmd_vel_msg->angular.z = deadman ? -msg->axes[0] * g_scale_angular : 0;
-    cmd_vel_pub->publish(cmd_vel_msg);
+    bool deadman = msg->buttons[4] != 0;
+    bool sent_zero = false;
```

Comment:
I think that you want `sent_zero` to be declared and initialized before coming into the main `while (rclcpp::ok())` loop (if I understand your intent correctly).